interfaces/IDeviceInfo.h

```diff
@@ -25,136 +25,271 @@
+            uint64_t totalRAM /* @brief Total installed system RAM memory (in bytes) */;
+            uint64_t freeRAM /* @brief Free system RAM memory (in bytes) */;
+            uint64_t totalSwap /* @brief Total swap space (in bytes) */;
+            uint64_t freeSwap /* @brief Swap space still available (in bytes) */;
+            string deviceName /* @brief Host name */;
+            string cpuLoad /* @brief Current CPU load (percentage) */;
+            CPULoadAvgs cpuLoadAvg /* @brief CPU load average */;
+            string serialNumber /* @brief Device serial number */;
+            string time /* @brief Current system date and time */;
+        };
+        struct Socket {
+            uint32_t total /* @brief Number of total */;
+            uint32_t open /* @brief Number of open */;
+            uint32_t link /* @brief Number of link */;
+            uint32_t exception /* @brief Number of exception */;
+            uint32_t shutdown /* @brief Number of shutdown */;
+            uint32_t runs /* @brief Number of runs */;
+        };
+
+        struct Address {
+            string name /* @brief Interface name */;
+            string mac /* @brief Interface MAC address */;
+            string /* @opaque */ ip /* @brief An array of Interface IP address */;
+        };
+        using IAddressIterator = RPC::IIteratorType<Address, ID_DEVICE_INFO_ADDRESSES>;
+
+        // @property
+        // @brief Retrieves Firware Information
+        virtual Core::hresult FirmwareVersion(FirmwareInfo& value /* @out */) const = 0;
+        // @property
+        // @brief Retrieves Device Info
+        virtual Core::hresult DeviceData(Device& value /* @out */) const = 0;
+        // @property
+        // @brief Retrieves SystemInfo
+        virtual Core::hresult SystemInfo(System& value /* @out */) const = 0;
+        // @property
+        // @brief Retrieves SocketInfo
+        virtual Core::hresult SocketInfo(Socket& value /* @out */) const = 0;
 
-        virtual uint32_t Configure(const PluginHost::IShell* service) = 0;
+        // @brief Retrieves AddressInfo
+        // @param addresses: An array of Interface address
+        virtual Core::hresult AddressInfo(IAddressIterator*& ip /* @out */) const = 0;
 
-        virtual uint32_t SerialNumber(string& value /* @out */) const = 0;
-        virtual uint32_t Sku(string& value /* @out */) const = 0;
-        virtual uint32_t Make(string& value /* @out */) const = 0;
-        virtual uint32_t ModelName(string& value/*@out*/) const = 0;
-        virtual uint32_t ModelYear(uint16_t& value/*@out*/) const = 0;
-        virtual uint32_t FriendlyName(string& value/*@out*/) const = 0;
-        virtual uint32_t DeviceType(string& value /* @out */) const = 0;
-        virtual uint32_t PlatformName(string& value/*@out*/) const = 0;
-        virtual uint32_t DistributorId(string& value /* @out */) const = 0;
+        /* @json:omit */
+        virtual Core::hresult Configure(const PluginHost::IShell* service) = 0;
```

Comment:
Lets drop this and use IConfiguration.h than we do not need the ommit and this is only an internal change for the plugin

Reply:
done